src/components/ReportActionItem/IOUAction.js

```diff
@@ -24,28 +31,53 @@ const propTypes = {
         /** The participants of this report */
         participants: PropTypes.arrayOf(PropTypes.string),
     }),
+
+    /** Session of currently logged in user */
+    session: PropTypes.shape({
+        email: PropTypes.string.isRequired,
+    }),
+
+    /** All of the personal details for everyone */
+    personalDetails: PropTypes.objectOf(personalDetailsPropType),
 };
 
 const defaultProps = {
     chatReport: {
         participants: [],
     },
+    session: {},
+    personalDetails: {},
 };
 
 const IOUAction = ({
     action,
     chatReportID,
     isMostRecentIOUReportAction,
+    chatReport,
+    session,
+    personalDetails,
 }) => {
     const launchDetailsModal = () => {
         Navigation.navigate(ROUTES.getIouDetailsRoute(chatReportID, action.originalMessage.IOUReportID));
     };
+
+    const sessionEmail = lodashGet(session, 'email', '');
+    const participantName = _.map(chatReport.participants, participant => lodashGet(personalDetails, [participant, 'firstName'], '')
+            || Str.removeSMSDomain(participant)).join(', ');
+
+    const currentUserName = lodashGet(personalDetails, [sessionEmail, 'firstName'], '')
+                        || Str.removeSMSDomain(sessionEmail);
+
     return (
         <>
             <IOUQuote
                 action={action}
                 shouldShowViewDetailsLink={Boolean(action.originalMessage.IOUReportID)}
                 onViewDetailsPressed={launchDetailsModal}
+                chatReport={chatReport}
+                currentUserName={currentUserName}
+                participantName={participantName}
+                sessionEmail={sessionEmail}
```

Comment:
I did that because `IOUAction` already subscribed to the `chatReport` from Onyx. To avoid two subscriptions: 1 in parent and second in child, I added the props to the child and added the extraction logic to parent.

Reply:
It looks like we are only using the `chatReport` prop in this component so that we can generate these other variables and then pass them to the child. Nothing else references the `chatReport` here.

And if you look at the previous diff this `chatReport` prop was not used at all although we were still subscribing to it. So, is there some reason for that or not? cc @Julesssss maybe you know the answer. If there is a reason we should leave a comment explaining why it's necessary.

Reply:
Yeah, that subscription is totally unnecessary. IIRC an early iteration of the Comp depended on `chatReport.participants`, so I must have forgotten to remove the subscription when refactoring.

Reply:
Let's move it down to IOUQuote 👍